packages/gazelle_cli/lib/commands/run/run_project.dart

```diff
@@ -0,0 +1,189 @@
+      return;
+    }
+    if (event.path.endsWith('.dart')) {
+      reload(reloader);
+    } else if(event.path.endsWith('pubspec.yaml')) {
+      reload(reloader);
+    } else {
+      print("Ignoring \${event.path} as it is not a dart file.");
+    }
+  }
+
+  lib.watch(recursive: true).listen(onModify);
+  pubspec.watch().listen(onModify);
+
+  ProcessSignal.sigint.watch().listen((event) {
+    reloader.stop();
+    exit(0);
+  });
+
+  stdin.transform(utf8.decoder).listen((event) async {
+    if (event.trim() == "r") {
+      reload(reloader);
+    }
+  });
+}
+
+/// This executionIndex is just to implement debouncing
+int executionIndex = 0;
+
+void reload(HotReloader reloader) async {
+  int tmp = ++executionIndex;
+  await Future.delayed(Duration(milliseconds: $timeout));
+
+  /// If the executionIndex has changed, then another restart was requested
+  /// within the timeout period, so we will let the other restart request
+  /// handle the rest and return
+  if (tmp != executionIndex) return;
+
+  reloader.reloadCode();
+}
+""";
+
+/// Represents an error during the running process of a project.
+class RunProjectError {
+  /// The error message.
+  final String message;
+
+  /// The error code.
+  final int errCode;
+
+  /// Creates a [RunProjectError].
+  RunProjectError(this.message, this.errCode);
+}
+
+/// Runs a Gazelle project.
+Future<void> runProject(String path, int timeout) async {
+  final projectDir = Directory(path);
+  if (!await projectDir.exists()) {
+    throw RunProjectError("Project not found!", 1);
+  }
+
+  final tmpDir =
+      Directory("${projectDir.absolute.path + Platform.pathSeparator}.tmp");
+  if (await tmpDir.exists()) {
+    await tmpDir.delete(recursive: true);
+  }
+  await tmpDir.create();
+
+  final tmpDirPath = tmpDir.absolute.path;
+
+  final projectName = path.split(Platform.pathSeparator).last;
+  await File("$tmpDirPath/pubspec.yaml")
+      .create(recursive: true)
+      .then((file) => file.writeAsString(_getPubspecTemplate(projectName)));
+
+  await File("$tmpDirPath/main_hot_reload.dart").create(recursive: true).then(
+      (file) => file.writeAsString(_getMainTemplate(projectName, timeout)));
+
+  final result = await Process.run(
+    "dart",
+    ["pub", "get"],
+    workingDirectory: tmpDirPath,
+  );
+
+  if (result.exitCode != 0) {
+    throw RunProjectError(result.stderr.toString(), result.exitCode);
+  }
+
+  Process? process = await startProcess(tmpDirPath);
+
+  stdinBroadcast.listen((event) async {
+    if (event.trim() == 'R') {
+      /// Hot Restart
+      process?.kill();
+      process = await startProcess(tmpDirPath);
+    } else {
+      /// Else send the input to the process, the process will be
+      /// responsible for if the event is 'r' then reload the code
+      process?.stdin.writeln(event);
+    }
+  });
+
+  ProcessSignal.sigint.watch().listen((event) {
+    process?.kill();
+
+    /// TODO: Delete the temporary directory here after the process is killed
```

Comment:
Is this a left over?

Reply:
Deleting the tmp project directory might not be necessary right now, but, we might need it to in the future. That is why I left it as it is.

Reply:
Removed TODO with 5dd54963a21867b1cb4952fe85bd97fd1b645507